src/Illuminate/Filesystem/Filesystem.php

```diff
@@ -789,4 +789,20 @@ public function cleanDirectory($directory)
     {
         return $this->deleteDirectory($directory, true);
     }
+
+    /**
+     * Format the number of bytes to a human-readable string.
+     *
+     * @param  int  $bytes
+     * @param  int  $precision
+     * @return string
+     */
+    public function bytesToHuman($bytes, $precision = 2)
+    {
+        for ($i = 0; ($bytes / 1024) > 0.9; $i++) {
+            $bytes /= 1024;
+        }
+
+        return round($bytes, $precision).' '.['B', 'KB', 'MB', 'GB', 'TB'][$i];
```

Comment:
That will not work for Petabyte: 

https://3v4l.org/aUKD7

So please use a fallback if we have more then the highest unit.

Reply:
Also: Maybe use `sprintf` for text output. This should imo be more readable.

Reply:
Sure thing! I could increase the lookup array to Yottabytes? I can of course also add a fallback, or both. What do you think is best?

Will use `sprintf` as well.

Reply:
Personally, I would like to see it expanded to yottabytes when it is irrelevant whether it is needed or not, it is there and the function can then. 

Additionally, you always need a fallback, as there can always be sizes you don't care about in the code (even if the integer is limited, you wan't always have a fallback).

So the easiest way for a fallback is a break within the loop, if we reach a certain point (or use a `for` loop with a break if `$bytes < 1024`).

Reply:
Thanks for the feedback!

7d046255e7ccd9f965c9bbc07345ccc8e8f907a2 adds a cap so that the iterator stops dividing when reaching the end of the units array which now also supports extreme values. Also uses `sprintf` as requested. Also adds tests for large numbers.

Let me know if you want me to squash the commits!